### DISCUSSION TITLE
SQL repo

### DISCUSSION THREAD

**Discussion by Tychodewaard** [dnnsoftware/Dnn.Platform#3686] [General] [answered]
In DNN there is a SQL console. Is it an idea to have one of the following?

- dropdown with queries. Now there is just 1 'new query'. Maybe only the safe to use ones like truncate table schedulehistory. Or with a lot of warnings. 
- link to a seperate repo where people place their SQL statements providing context and explanation

**Comment by sleupold** [comment, marked as answer]
@Tychodewaard 

- once you saved a query, the "query" input fields becomes a dropdown list, offering all saved queries.
- I already thought about a repo, but it would need to be version specific, as data structures may change from DNN (or extension) version to the next - e.g. Truncate Eventlog changed significantly in DNN 7.4.0. I see a big security risk...
